packages/keyring-controller/src/KeyringController.test.ts

```diff
@@ -563,6 +563,31 @@ describe('KeyringController', () => {
               },
             );
           });
+
+          it('should remove encryptionKey and encryptionSalt from state when cacheEncryptionKey is false', async () => {
+            await withController(
+              { cacheEncryptionKey: false },
+              async ({ controller }) => {
+                await controller.createNewVaultAndKeychain(password);
+
+                expect(controller.state).not.toHaveProperty('encryptionKey');
+                expect(controller.state).not.toHaveProperty('encryptionSalt');
+              },
+            );
+          });
+
+          cacheEncryptionKey &&
+            it('should set encryptionKey and encryptionSalt in state', async () => {
+              // TODO: Either fix this lint violation or explain why it's necessary to ignore.
+              // eslint-disable-next-line @typescript-eslint/no-floating-promises
+              withController(
+                { cacheEncryptionKey },
+                async ({ initialState }) => {
+                  expect(initialState.encryptionKey).toBeDefined();
```

Comment:
The `controller.createNewVaultAndKeychain` call is missing from this test

Reply:
Oh wait, it's probably in the setup isn't it

Reply:
But wait, this block of tests is under `when there is no existing vault`, so we should be skipping the vault creation for these tests

Reply:
For these first two I'd recommend using the `{ skipVaultCreation: true },` option and calling `createNewVaultAndKeychain` explicitly, as most of the tests in this block do.

For the one below on line 627, I think we're just missing the `controller.createNewVaultAndKeychain` call.

Reply:
> For these first two I'd recommend using the `{ skipVaultCreation: true },` option and calling `createNewVaultAndKeychain` explicitly, as most of the tests in this block do.
> 
> For the one below on line 627, I think we're just missing the `controller.createNewVaultAndKeychain` call.

Regarding the test on line 627. That test existed as is prior to my change but I think its good to add this for consistency and clarity.